command/agent/cache/handler.go

```diff
@@ -81,53 +81,74 @@ func processTokenLookupResponse(ctx context.Context, logger hclog.Logger, useAut
 		return nil
 	}
 
-	// Strip-off namespace related information from the request and get the
-	// relative path of the request.
 	_, path := deriveNamespaceAndRevocationPath(req)
-	if path == vaultPathTokenLookupSelf {
-		logger.Info("stripping auto-auth token from the response", "path", req.Request.URL.Path, "method", req.Request.Method)
-		secret, err := api.ParseSecret(bytes.NewBuffer(resp.ResponseBody))
-		if err != nil {
-			return fmt.Errorf("failed to parse token lookup response: %v", err)
+	switch path {
+	case vaultPathTokenLookupSelf:
```

Comment:
Brian's comment got lost but is still valid: this will do nothing for lookupself...you want `case vaultPathTokenLookupSelf, vaultPathTokenLookup`.